tests/Doctrine/Tests/Common/Cache/MemcachedCacheTest.php

```diff
@@ -49,6 +49,31 @@ public function testGetMemcachedReturnsInstanceOfMemcached()
         $this->assertInstanceOf('Memcached', $this->_getCacheDriver()->getMemcached());
     }
 
+    public function testContainsWithKeyWithFalseAsValue()
+    {
+        $testKey = __METHOD__;
+        $driver = $this->_getCacheDriver();
+        $reflection = new \ReflectionClass($driver);
+        $method = $reflection->getMethod('getNamespacedId');
+        $method->setAccessible(true);
+        $testKeyNS = $method->invokeArgs($driver, [$testKey]);
+        $this->memcached->set($testKeyNS, false);
```

Comment:
This will test the code works even if the item was cached in a previous run of the code and not only during a single executing of it. I you think it holds no value I can refactor. 

TBH we are no longer using the caching layer of Doctrine at all (we use a custom one because of other issues we have with it) so I don't want to spend much more time/effort on this PR. Please let me know if you see more issue with this so I can address those in one go.